command/tls_cert_create.go

```diff
@@ -294,3 +300,46 @@ func (c *TLSCertCreateCommand) Run(args []string) int {
 
 	return 0
 }
+
+func recordPreparation(certType string, regionName string, domain string, ipAddresses []net.IP) ([]net.IP, []string, string, []x509.ExtKeyUsage, string) {
+	var (
+		DNSNames                []string
+		extKeyUsage             []x509.ExtKeyUsage
+		name, regionUrl, prefix string
+	)
+	if certType == "server" {
+		extKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth}
+		ipAddresses = append(ipAddresses, net.ParseIP("127.0.0.1"))
+	} else if certType == "client" {
+		extKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth}
+		ipAddresses = append(ipAddresses, net.ParseIP("127.0.0.1"))
+	} else if certType == "cli" {
+		extKeyUsage = []x509.ExtKeyUsage{}
+	}
```

Comment:
The `extKeyUsage` and `ipAddresses` function values seem to have little to do with the rest of this code. I'd break this out to its own function. Also, is this correct if `ipAddresses` already contains `"127.0.0.1"`?

Reply:
I could refactor the extKeyUsage to be if server || client and then an empty one for the cli case.